lib/vault/client.rb

```diff
@@ -392,6 +392,8 @@ def error(response)
 
       # Use the correct exception class
       case response
+      when Net::HTTPPreconditionFailed
+        raise MissingRequiredStateError.new
```

Comment:
If the 412 response from Vault actually has the context that you hardcoded in the message there “The performance standby node…“, generally speaking, I’d probably match the signatures and return the message from Vault directly.  That also avoids the slight weirdness of raising an exception in a method that sets everything to then raise an exception.

So you could probably catch the 412 and then let the existing logic work for you in HTTPError:

```
The Vault server at `#{address}' responded with a #{code}.
Any additional information the server supplied is shown below:
#{errors.join("\n").rstrip}
Please refer to the documentation for help.
```

Which I’m assuming would end up like this:

```
The Vault server at `#{address}' responded with a 412.
Any additional information the server supplied is shown below:

The performance standby node does not yet have the most recent....

Please refer to the documentation for help.
```

What does Vault actually return for this error? 

Otherwise, if Vault doesn’t actually return anything useful beyond the status code, what you have is probably fine.

Reply:
Vault seems to not return any response body for this error, just the status code.